boards/native/board_init.c

```diff
@@ -20,7 +20,57 @@
 
 #ifdef MODULE_MTD
 #include "mtd_native.h"
+
+mtd_native_dev_t mtd0_dev = {
+    .dev = {
+        .driver = &native_flash_driver,
+        .sector_count = MTD_SECTOR_NUM,
+        .pages_per_sector = MTD_SECTOR_SIZE / MTD_PAGE_SIZE,
+        .page_size = MTD_PAGE_SIZE,
+    },
+    .fname = MTD_NATIVE_FILENAME,
+};
+
+mtd_dev_t *mtd0 = (mtd_dev_t *)&mtd0_dev;
```

Comment:
What about using `container_of` here?

Reply:
`container_of` is the opposite direction. We could do `&mtd0_dev.dev` here.

Unfortunately there is no agreement between MTD drivers how the `mtd_dev_t` the inherit should be called, only that it has to be the first element.

So for `mtd_native_dev_t` it's `dev` and for `mtd_spi_nor_t` it's `base``.

I guess the cast is less confusing.

Reply:
> container_of is the opposite direction. We could do &mtd0_dev.dev here.

oops, sorry, you are right, I got confused because I was thinking in the beginning maybe it made sense to use `container_of` in `VFS_AUTO_MOUNT` to avoid the blind cast.

Reply:
> only that it has to be the first element.

This is the only part that makes me a bit nervous. Just relying on this convention may not be the best.

Reply:
Yes but this is unrelated to this PR :wink:

Reply:
```suggestion
mtd_dev_t *mtd0 = &mtd0_dev.base;
```

Now that you have renamed it to `base` in `mtd_native_dev_t`, this should work, shouldn't it?